### PR TITLE
feat(connector): [BOA/Cybersource] Pass commerce indicator using card network for apple pay

### PR DESCRIPTION
**Pull request by deepanshu-iiitu**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Apple pay transactions via BOA and Cybersource require a specific field `commerce_indicator` which is based on the card network type used during apple pay transactions.
https://docs.cybersource.com/content/dam/documentation/en/apple-pay/fdiglobal/so/applepay-so-fdiglobal.pdf

This data is now being passed using the card network data being shared from SDK during apple pay payments.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

https://github.com/juspay/hyperswitch/issues/3783

## How did you test it?

Apple pay payments need to be tested using simplified and manual flow for connectors BOA and Cybersource.
image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/bankofamerica/transformers.rs [hunk 1]
```diff
@@ -330,21 +330,32 @@ impl
...
     fn from(
-        (item, solution): (
+        (item, solution, network): (
             &BankOfAmericaRouterData<&types::PaymentsAuthorizeRouterData>,
             Option<PaymentSolution>,
+            Option<String>,
         ),
     ) -> Self {
+        let commerce_indicator = match network {
+            Some(card_network) => match card_network.as_str() {
```
Comment (sai-harsha-vardhan) [inline crates/router/src/connector/bankofamerica/transformers.rs]: Convert these to lower case and then match. And need not construct String in every case, can do .to_string() while returning

File: crates/router/src/core/payments.rs [hunk 2]
```diff
@@ -1129,6 +1129,8 @@ where
             .parse_value::<router_types::ApplePayPredecryptData>("ApplePayPredecryptData")
             .change_context(errors::ApiErrorResponse::InternalServerError)?;
 
+        logger::debug!(?apple_pay_predecrypt);
```
Comment (sai-harsha-vardhan) [inline crates/router/src/core/payments.rs:1132]: Why is this required?

File: crates/router/src/connector/bankofamerica/transformers.rs (hunk 1 above)
Comment (deepanshu-iiitu) [reply crates/router/src/connector/bankofamerica/transformers.rs]: Done

**Review by sai-harsha-vardhan — APPROVED** [review]

**Review by srujanchikke — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by ShankarSinghC — APPROVED** [review]
